src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -684,6 +717,63 @@ private static bool InitRTNoCacheSupported()
                 "NoCache");
         }
 
+        private void RTServerCertificateCallback(RTHttpBaseProtocolFilter sender, RTHttpServerCustomValidationRequestedEventArgs args)
+        {
+            // Convert WinRT certificate to .NET certificate.
+            X509Certificate2 serverCert = ConvertCertificate(args.ServerCertificate);
+
+            // Create .NET X509Chain from the WinRT information. We need to rebuild
+            // the chain since WinRT only gives us an array of intermediate certificates.
+            var serverChain = new X509Chain();
```

Comment:
Probably. I was  afraid to do it because I didn't remember the behavior contract in terms of .NET Framework.  I.e., what happens if a user-provided callback wants to save those values for later inspection, i.e. cert-pinning scenarios?

I did notice that we dispose of them in WinHttpHandler.

```c#
finally
{
    if (chain != null)
    {
        chain.Dispose();
    }

    serverCertificate.Dispose();
 }
```
But that might not be correct. We might need to stop disposing of them in WinHttpHandler and *Nix. Need to research what the behavior is on .NET Framework.